pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -365,32 +380,31 @@ def solve(self, model, **kwds):
             # Set of MIP iterations for which cuts were generated in ECP
             lin.mip_iters = Set(dimen=1)
 
-            nonlinear_constraints = [c for c in MindtPy.constraint_list if
-                                     c.body.polynomial_degree() not in (1, 0)]
-            lin.nl_constraint_set = RangeSet(
-                len(nonlinear_constraints),
-                doc="Integer index set over the nonlinear constraints")
-            feas.constraint_set = RangeSet(
-                len(MindtPy.constraint_list),
-                doc="integer index set over the constraints")
-
-            # # Mapping Constraint -> integer index
-            # MindtPy.feas_map = {}
-            # # Mapping integer index -> Constraint
-            # MindtPy.feas_inverse_map = {}
-            # # Generate the two maps. These maps may be helpful for later
-            # # interpreting indices on the slack variables or generated cuts.
-            # for c, n in zip(MindtPy.constraint_list, feas.constraint_set):
-            #     MindtPy.feas_map[c] = n
-            #     MindtPy.feas_inverse_map[n] = c
-
-            # Create slack variables for OA cuts
+            if config.feasibility_norm == 'L1' or config.feasibility_norm == 'L2':
+                feas.nl_constraint_set = Set(initialize=[i for i, constr in enumerate(MindtPy.constraint_list, 1) if
+                                                         constr.body.polynomial_degree() not in (1, 0)],
+                                             doc="Integer index set over the nonlinear constraints."
+                                             "The set corresponds to the index of nonlinear constraint in constraint_set")
+                # Create slack variables for feasibility problem
+                feas.slack_var = Var(feas.nl_constraint_set,
+                                     domain=NonNegativeReals, initialize=1)
+            else:
+                feas.slack_var = Var(domain=NonNegativeReals, initialize=1)
+
+                # # Mapping Constraint -> integer index
```

Comment:
Is there any reason we need these maps here? If not we can just remove these comments.

Reply:
These are legacy from my original implementation, back when I put in unnecessary maps in case I would need to query them. They can probably be removed.